### PR TITLE
Flag sexp descriptions

### PR DESCRIPTION
**Pull request by MjnMixael**
As we create more and more ship/wing flags, it becomes harder to remember the nuances of what each one does while FREDing. This PR creates a framework for displaying in the Event Editor helpbox a description of what each flag actually does when it's the highlighted node. It does add some additional stuff to manage with each flag, but other methods I tried required larger changes in unrelated code locations that I didn't want to touch just to make this work.

Basically we're using the node text to lookup the flag id then using the flag id to grab the flag description. I did it this way because, if you all are like me, you "find all" on a previously existing flag when creating a new one to make sure you add the necessary code in all the right places.

### CODE REVIEW

File: code/mission/missionparse.cpp [hunk 1]
```diff
@@ -338,6 +338,70 @@ flag_def_list_new<Mission::Parse_Object_Flags> Parse_object_flags[] = {
...
+    { Mission::Parse_Object_Flags::Knossos_warp_in,					"Ship uses the special Knossos warp in animation."},
+    { Mission::Parse_Object_Flags::SF_Vaporize,						"Causes a ship to vanish (no deathroll, no debris, no explosion) when destroyed."},
+    { Mission::Parse_Object_Flags::SF_Stealth,						"If set, the ship can't be targeted, is invisible on radar, and is ignored by AI unless firing."},
+    { Mission::Parse_Object_Flags::SF_Friendly_stealth_invis,		"If set, the ship can't be targeted even by ships on the same team."},
+    { Mission::Parse_Object_Flags::SF_Dont_collide_invis,			"Will cause polygons with an invisible texture to stop colliding with objects."},
+    { Mission::Parse_Object_Flags::SF_Primitive_sensors,			"Targets will only be a blip on the radar. Ships cannot targeted and aspect lock cannot be used."},
+    { Mission::Parse_Object_Flags::SF_No_subspace_drive,			"Will not allow a ship to jump into subspace."},
+    { Mission::Parse_Object_Flags::SF_Nav_carry_status,				"This ship autopilots with the player."},
+    { Mission::Parse_Object_Flags::SF_Affected_by_gravity,			"Ship is affected by mission gravity."},
```
Comment (Baezon) [inline code/mission/missionparse.cpp:372]: This is a vestigal flag that does nothing, despite the name it has nothing to do with the recently added mission gravity. It really should be hidden where possible to avoid confusion tbh...

File: code/mission/missionparse.cpp [hunk 2]
```diff
@@ -338,6 +338,70 @@ flag_def_list_new<Mission::Parse_Object_Flags> Parse_object_flags[] = {
...
+    { Mission::Parse_Object_Flags::OF_Beam_protected,				"Turrets with beam weapons will ignore and not attack ship."},
+    { Mission::Parse_Object_Flags::OF_Flak_protected,				"Turrets with flak weapons will ignore and not attack ship."},
+    { Mission::Parse_Object_Flags::OF_Laser_protected,				"Turrets with laser weapons will ignore and not attack ship."},
+    { Mission::Parse_Object_Flags::OF_Missile_protected,			"Turrets with missile weapons will ignore and not attack ship."},
+    { Mission::Parse_Object_Flags::SF_Guardian,						"Ship health cannot be reduced below 1."},
+    { Mission::Parse_Object_Flags::Knossos_warp_in,					"Ship uses the special Knossos warp in animation."},
+    { Mission::Parse_Object_Flags::SF_Vaporize,						"Causes a ship to vanish (no deathroll, no debris, no explosion) when destroyed."},
+    { Mission::Parse_Object_Flags::SF_Stealth,						"If set, the ship can't be targeted, is invisible on radar, and is ignored by AI unless firing."},
+    { Mission::Parse_Object_Flags::SF_Friendly_stealth_invis,		"If set, the ship can't be targeted even by ships on the same team."},
```
Comment (Baezon) [inline code/mission/missionparse.cpp:367]: '...if it is *also* flagged as stealth.'

**Review by Baezon — CHANGES_REQUESTED** [review]
I can't comment much on the FRED changes, just some small corrections regarding the new documentation.

File: code/mission/missionparse.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/mission/missionparse.cpp]: I'm not sure if the alter-ship-flag hides that one or not, but that is outside the scope of what I'm doing here. I agree that a way to deprecate and hide flags would be idea. For now, I changed the description to reflect that it does nothing.

File: code/mission/missionparse.cpp (hunk 2 above)
Comment (MjnMixael) [reply code/mission/missionparse.cpp]: fixed

**Review by Baezon — APPROVED** [review]
Still would benefit from a review from someone better versed in sexps and FRED.

**Review by Goober5000 — APPROVED** [review]
Yup, this looks good, and is a useful feature.  Some comments on the descriptions, but the functionality looks solid.

File: code/ship/ship.cpp [hunk 3]
```diff
@@ -559,6 +590,15 @@ wing_flag_name Wing_flag_names[] = {
 	{ Wing_Flags::Same_departure_warp_when_docked,	"same-departure-warp-when-docked" }
 };
 
+wing_flag_description Wing_flag_descriptions[] = {
+	{ Wing_Flags::No_arrival_music,					"Arrival music will not be played when this wing arrives."},
+	{ Wing_Flags::No_arrival_message,				"Arrival messages will not be sent when this wing arrives."},
+	{ Wing_Flags::No_arrival_warp,					"No ship in the wing will use a warp effect upon arrival."},
+	{ Wing_Flags::No_departure_warp,				"No ship in the wing will use a warp effect upon departure."},
+	{ Wing_Flags::Same_arrival_warp_when_docked,	"All ships in the wing will use a single warp effect on arrival if docked."},
+	{ Wing_Flags::Same_departure_warp_when_docked,	"All ships in the wing will use a single warp effect on departure if docked."}
```
[the remark this answers is not in the document]
Comment (MjnMixael) [reply code/ship/ship.cpp]: all done

File: code/mission/missionparse.cpp [hunk 4]
```diff
@@ -383,17 +383,17 @@ parse_object_flag_description<Mission::Parse_Object_Flags> Parse_object_flag_des
...
     { Mission::Parse_Object_Flags::SF_Scramble_messages,			"All messages sent from this ship appear scrambled."},
     { Mission::Parse_Object_Flags::OF_No_collide,					"Ship cannot be collided with."},
     { Mission::Parse_Object_Flags::SF_No_disabled_self_destruct,	"Ship will not self-destruct after 90 seconds if engines or weapons destroyed."},
     { Mission::Parse_Object_Flags::SF_Hide_mission_log,				"Mission log events generated for this ship will not be viewable."},
-    { Mission::Parse_Object_Flags::SF_Same_arrival_warp_when_docked,"Docked ships use the same warp effect upon arrival."},
-    { Mission::Parse_Object_Flags::SF_Same_departure_warp_when_docked,"Docked ship use the same warp effect upon departure."},
+    { Mission::Parse_Object_Flags::SF_Same_arrival_warp_when_docked,"Docked ships use the same warp effect size upon arrival as if they were not docked instead of the enlarged arggregate size."},
+    { Mission::Parse_Object_Flags::SF_Same_departure_warp_when_docked,"Docked ship use the same warp effect size upon departure as if they were not docked instead of the enlarged arggregate size."},
```
Comment (Goober5000) [inline code/mission/missionparse.cpp:396]: aggregate

File: code/ship/ship.cpp [hunk 5]
```diff
@@ -595,8 +595,8 @@ wing_flag_description Wing_flag_descriptions[] = {
 	{ Wing_Flags::No_arrival_message,				"Arrival messages will not be sent when this wing arrives."},
 	{ Wing_Flags::No_arrival_warp,					"No ship in the wing will use a warp effect upon arrival."},
 	{ Wing_Flags::No_departure_warp,				"No ship in the wing will use a warp effect upon departure."},
-	{ Wing_Flags::Same_arrival_warp_when_docked,	"All ships in the wing will use a single warp effect on arrival if docked."},
-	{ Wing_Flags::Same_departure_warp_when_docked,	"All ships in the wing will use a single warp effect on departure if docked."}
+	{ Wing_Flags::Same_arrival_warp_when_docked,	"All ships in the wing will use the same warp effect size upon arrival as if they were not docked instead of the enlarged arggregate size."},
+	{ Wing_Flags::Same_departure_warp_when_docked,	"All ships in the wing will use the same warp effect size upon departure as if they were not docked instead of the enlarged arggregate size."}
```
Comment (Goober5000) [inline code/ship/ship.cpp:599]: aggregate

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Cool.  Two more

File: code/ship/ship.cpp (hunk 5 above)
Comment (MjnMixael) [reply code/ship/ship.cpp]: done

**Review by Goober5000 — APPROVED** [review]
